### PR TITLE
[CouchDB] Using multiple databases 

### PR DESCRIPTION
**Pull request by xlecours**
This pull request should allow the CouchDB object to access different databases. It transform the singleton in a multiton where each instance point to a specific database. The default value is the DTQ (CouchDB config setting). It also alter the NDB_Factory class transforming the _couchdb variable in an array.

* This is the initial commit (code using this is on the way)

### CODE REVIEW

**Comment by ridz1208** [comment]
@xlecours if this does what i think it does... It would be my christmas present this year

**Comment by xlecours** [comment]
@jstirling91 This might affect the DQT (but shouldn't)

File: tools/CouchDB_Import_RadiologicalReview.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ class CouchDBRadiologicalReviewImporter {
 
     function __construct() {
         $this->SQLDB = Database::singleton();
```
Comment (ridz1208) [inline tools/CouchDB_Import_RadiologicalReview.php]: add the \

File: tools/CouchDB_Import_Instruments.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ class CouchDBInstrumentImporter
     function __construct()
     {
         $this->SQLDB   = Database::singleton();
```
Comment (ridz1208) [inline tools/CouchDB_Import_Instruments.php]: add the \

File: tools/CouchDB_Import_Demographics.php [hunk 3]
```diff
@@ -102,7 +102,7 @@ class CouchDBDemographicsImporter {
 
     function __construct() {
         $this->SQLDB = Database::singleton();
```
Comment (ridz1208) [inline tools/CouchDB_Import_Demographics.php]: add the \

File: php/libraries/CouchDB.class.inc [hunk 4]
```diff
@@ -165,22 +175,43 @@ class CouchDB
...
+    static function &getInstance($database = null)
     {
-        static $db = null;
-        if ($db === null) {
-            $db = new CouchDB();
+        if (empty(self::$_instances[$database])) {
+
+            if (empty($database)) {
+                // Default to DQT
+                $config   = NDB_Config::singleton();
```
Comment (ridz1208) [inline php/libraries/CouchDB.class.inc]: add the \ ???

File: tools/CouchDB_Import_Demographics.php (hunk 3 above)
Comment (driusan) [reply tools/CouchDB_Import_Demographics.php]: It's not in a namespace, I don't think it'll make a difference?

**Pull request merged**